circle.yml

```diff
@@ -1,28 +1,48 @@
-machine:
-  pre:
-     # install updated docker
-     - curl -sSL https://s3.amazonaws.com/circle-downloads/install-circleci-docker.sh | bash -s -- 1.10.0
-     - pip install docker-compose
-  services:
-    - docker
-  python:
-    version: 2.7.11
+version: 2
+jobs:
+  build:
+    docker:
+      # base image for primary container
+      - image: circleci/python:3.6.1
 
-dependencies:
-  override:
-    # create a version.json per https://github.com/mozilla-services/Dockerflow/blob/master/docs/version_object.md
-    # we don't dockerflow this, but it's handy to have a build artifact
-    - >
-        printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n'
-        "$CIRCLE_SHA1"
-        "$CIRCLE_TAG"
-        "$CIRCLE_PROJECT_USERNAME"
-        "$CIRCLE_PROJECT_REPONAME"
-        "$CIRCLE_BUILD_URL" > version.json
-    - cp version.json $CIRCLE_ARTIFACTS
+    working_directory: ~/pigeon
 
-test:
-  pre:
-    - chmod -R 777 $CIRCLE_TEST_REPORTS
-  override:
-    - make test
+    steps:
+      - checkout
+
+      - run:
+          name: Create version.json
+          command: |
+            # create a version.json per https://github.com/mozilla-services/Dockerflow/blob/master/docs/version_object.md
+            # we don't dockerflow this, but it's handy to have a build artifact
+            printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n' \
+            "$CIRCLE_SHA1" \
+            "$CIRCLE_TAG" \
+            "$CIRCLE_PROJECT_USERNAME" \
+            "$CIRCLE_PROJECT_REPONAME" \
+            "$CIRCLE_BUILD_URL" > version.json
+
+      - store_artifacts:
+          path: version.json
+
+      - setup_remote_docker
+
+      - run:
+          name: Build images
+          command: make build
+
+      - run:
+          name:  Run tests
+          # Have to use docker so we don't mount a volume which doesn't
+          # work in Circle CI
+          command: |
+            docker-compose up -d rabbitmq
+            docker network ls
+            docker run \
+              --rm \
+              --workdir=/app \
+              --network=pigeon_default \
+              --link=pigeon_rabbitmq_1 \
+              --env-file=docker.env \
+              pigeon_lambda:latest \
+              py.test
```

Comment:
"WTF IS THIS!?" you might ask.

Friend, let me tell you of a time before the age of history in which there was a young king who had trouble remembering where he put his glasses. He would shout and rant and yell about how it was hard to see anything in the castle and his loyal subjects would bow deeply in terror--unsure whether to tell the king he probably left them in the royal "throne room" again.

One day, a peasant from another kingdom who was traveling through selling sporks to all the gentry found himself under the ranting gaze of the de-spectacled young king. He bowed low, but then looked up, and said, "Sire. I know you can't see mounted volumes. Perhaps we should run docker manually rather than using the docker-compose interface because the files in question are already in the image in /app. It's the best thing to do even if it seems like you're hardcoding a bunch of values. Also, your glasses are probably in the royal throneroom. I heard someone mention you often leave them there."

The room was deathly silent. So silent you could hear the gentle sounds of dust drifting lazily in the morning air. The king looked down at this peasant and then said, "Oh, yes. I suppose they are. That docker thing bit me last year and I spent several hours re-discovering it. I wish I had written a silly story then so that I might remember better today."

He then wandered off down the hall instructing his royal treasurer to give everyone in his kingdom a million gold and a cat. Everyone lived happily ever after. The end.